### ISSUE TITLE
Add Delete permissions 

### ISSUE THREAD

**Issue by annavik** [RolnickLab/antenna#424]
For users with "staff status", seems delete permission is missing. For research sites, I also tried to manually assign this permission from admin portal, but it did not help. For now, I'm allowing delete from the interface if the update permission is present, but would be nice to change.

<img width="565" alt="Skärmavbild 2024-06-12 kl  12 08 50" src="https://github.com/RolnickLab/ami-platform/assets/11680517/ee111272-9a92-47cc-812a-6f8a62de0a63">

**Comment by mihow** [comment]
This is part of the work for issue #219 and will likely be closed with #693

**Comment by annavik** [comment]
Yes, this will be fixed! The frontend code is updated to consider actual delete permissions in #693 and this worked well during testing as well :)

**Comment by mohamedelabbas1996** [comment]
This should be closed after merging #693
